src/coreclr/debug/dbgutil/elfreader.cpp

```diff
@@ -314,7 +326,7 @@ ElfReader::GetStringAtIndex(int index, std::string& result)
     return true;
 }
 
-#ifdef HOST_UNIX
+#if defined(HOST_UNIX) && !defined(TARGET_HAIKU)
```

Comment:
I am just curious here - are we disabling this for Haiku because its executable files don't support the stuff the two methods below try to extract or does it just require more work for Haiku?

Reply:
Don't know about the executable files themselves, but Haiku headers do not have `struct r_debug` and `struct link_map`.

Reply:
Ok, got it.